test/unit/mailers/post_office_test.rb

```diff
@@ -20,8 +20,9 @@ class PostOfficeTest < ActionMailer::TestCase
   end
 
   test 'message_notification is graceful when recipient account is deleted' do
-    message = Message.create! sender: @provider, to: [@buyer], subject: "I am subject"
-    @buyer.destroy!
+    message = Message.create! sender: @provider, subject: "I am subject"
+    recipient = FactoryBot.create(:received_message, receiver: @buyer, message:)
+    recipient.update_column(:receiver_id, 0)
```

Comment:
why `0`?

Reply:
This simulates the situation when the receiver account has been deleted, because `MessageRecipient.receiver_id` is pointing to an account that doesn't exist. No need to be `0` in particular, it could be any number as long as we are sure it's not an existing account. But `0` works fine.

Reply:
Thank you @jlledom and @akostadinov